Install test api-blocks package and grpc-web

kimlisa opened the pull request:
task #16 

- rpc calls returned with errors will auto retry of up to 3 times

### Test out grpc-web:
Starting at root of `hwsc-frontend`

```
$ npm run serve
$ cd docker
$ docker-compose up
```

Might need to run:
```
$ docker rm $(docker ps -a -q)
$ docker rmi $(docker images -q)
```
Successful response from server:
![Screen Shot 2019-05-21 at 11 58 52 AM](https://user-images.githubusercontent.com/43280172/58139674-a9a45f00-7bf0-11e9-9584-da8512de29c8.png)

Successful response from browser
![Screen Shot 2019-05-21 at 11 58 46 AM](https://user-images.githubusercontent.com/43280172/58139683-b32dc700-7bf0-11e9-929d-89c343db0467.png)

kimlisa committed:
Install test api-blocks package

kimlisa committed:
Update docker-compose to include envoy proxy

kimlisa committed:
Rename envoyproxy file names

kimlisa committed:
Add ts types for variables related to proto

faraonc commented on src/App.vue:
```diff
@@ -19,6 +22,24 @@ import HelloI18n from './components/HelloI18n.vue';
+    const request: AppGatewayServiceRequest = new AppGatewayServiceRequest();
+
+    // TODO need to make some sort of metadata interface as a type
+    const metadata = { 'custom-header-1': 'value1' };
```
To authenticate you have to send a metadata to `app-gateway-svc`.
Check the following:
- review the epic especially step 3 in https://hwsc-org.github.io/wikis/app-gateway-svc/epics.html#authenticateuser
- https://github.com/hwsc-org/hwsc-api-blocks/blob/master/protobuf/hwsc-app-gateway-svc/hwsc-app-gateway-svc.js#L24
- note the base 64 encoding
- note the email & password from the 
config https://hwsc.slack.com/archives/CGF26TN1Z/p1555385177000700
- the email & password only allows `CreateUser`

faraonc commented on docker/envoy_proxy/envoy.yaml:
```diff
@@ -0,0 +1,49 @@
+                  address:
+                    socket_address:
+                      address: dev-app-gateway-svc
+                      port_value: 50055
```
I have no idea how this envoy works. I am gonna have to catch up later on this after I setup ElasticSearch.

faraonc commented on src/App.vue:
```diff
@@ -19,6 +22,24 @@ import HelloI18n from './components/HelloI18n.vue';
+  client: AppGatewayServiceClient = undefined as any;
+
+  created() {
+    this.client = new AppGatewayServiceClient('http://localhost:50056', null, null);
```
We should check it out if we can add an authentication header in `new AppGatewayServiceClient('http://localhost:50056', null, null);` instead of adding it as metadata per GRPC calls https://github.com/hwsc-org/hwsc-frontend/pull/21/files#r285904231

kimlisa committed:
Update README and env.development.local file

faraonc reviewed (dismissed):
This PR looks done to me thus task https://github.com/hwsc-org/hwsc-frontend/issues/16 is *completed*. For this task we only need for services to be able to communicate.

The following lines should not be in `App.vue` and `GetStatus` should be called in the `mutations` property of vuex because it is an http client `app-gateway-svc` client. Consider adding a temporary button to `dispatch` grpc `GetStatus`
```
import { AppGatewayServiceRequest } from '@hwsc/hwsc-api-blocks-lisa-test/protobuf/hwsc-app-gateway-svc/app/hwsc-app-gateway-svc_pb';
import { AppGatewayServiceClient } from '@hwsc/hwsc-api-blocks-lisa-test/protobuf/hwsc-app-gateway-svc/app/hwsc-app-gateway-svcServiceClientPb';
    this.client = new AppGatewayServiceClient('http://localhost:50056', null, null);

```

This means that you should be working on task https://github.com/hwsc-org/hwsc-frontend/issues/22

kimlisa committed:
Make a successfull getStatus call

- create docker/README that includes docker commands and notes
- updated root/README: moved running docker-compose under CLI commands
- updated yaml file to include authorization header

kimlisa committed:
Replace hard-coded values with system's env vars
